sentry/src/main/java/io/sentry/SentryWrapper.java

```diff
@@ -12,24 +12,26 @@
  *   <li>{@link Supplier}
  * </ul>
  *
- * that forks the current scope before execution and restores it afterwards. This prevents reused
- * threads (e.g. from thread-pools) from getting an incorrect state.
+ * that forks the current scope(s) before execution and restores previous state afterwards. Which
+ * scope(s) are forked, depends on the method used here. This prevents reused threads (e.g. from
+ * thread-pools) from getting an incorrect state.
  */
+// TODO [HSM] only deliver isolated variant as default for now
```

Comment:
To avoid confusion we should probably just start off by choosing the best default for all scope forking which in this case should mean forking both isolation and current scope. We can always come back and add other variants if requested by customers.